mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextEditorImpl.java

```diff
@@ -43,235 +45,266 @@
+
+        // If there are more than one lines, there is a line separator
+        lineSeparatorExists = textArea.getLineCount() > 1;
+
+        // Move cursor to the top
+        textArea.setCaretPosition(0);
+    }
+
+    void write(Writer writer) throws IOException {
+        Document document = textArea.getDocument();
+
+        // According to the documentation in DefaultEditorKit, the line separator is set to be as the system property
+        // if no other line separator exists in the file, but in practice it is not, so this is a workaround for it
+        if (!lineSeparatorExists)
+            document.putProperty(DefaultEditorKit.EndOfLineStringProperty, System.getProperty("line.separator"));
+
+        try {
+            textArea.getUI().getEditorKit(textArea).write(new BufferedWriter(writer), document, 0, document.getLength());
+        } catch (BadLocationException e) {
+            throw new IOException(e.getMessage());
+        }
+    }
+
+
+    public void setSyntaxBasedOnFilename(String fileExtension) {
+        // TODO use Apache Tika or some other way to work out mime-type (java built-in options suck)
+        String defaultMimeType = SyntaxConstants.SYNTAX_STYLE_NONE;
+        Map<String, String> extToMime = new HashMap<String, String>() {{
+            put("java", SyntaxConstants.SYNTAX_STYLE_JAVA);
+            put("js", SyntaxConstants.SYNTAX_STYLE_JAVASCRIPT);
+            put("gradle", SyntaxConstants.SYNTAX_STYLE_GROOVY);
+            put("c", SyntaxConstants.SYNTAX_STYLE_C);
+            put("cpp", SyntaxConstants.SYNTAX_STYLE_CPLUSPLUS);
+            put("html", SyntaxConstants.SYNTAX_STYLE_HTML);
+            put("htm", SyntaxConstants.SYNTAX_STYLE_HTML);
+        }};
+
+        textArea.setSyntaxEditingStyle(extToMime.getOrDefault(fileExtension.toLowerCase(), defaultMimeType));
+    }
+
+    //////////////////////////////////
+    // ThemeListener implementation //
+    //////////////////////////////////
+
+    /**
+     * Receives theme color changes notifications.
+     */
+    public void colorChanged(ColorChangedEvent event) {
+        switch (event.getColorId()) {
+            case Theme.EDITOR_FOREGROUND_COLOR:
+                textArea.setForeground(event.getColor());
+                break;
+
+            case Theme.EDITOR_BACKGROUND_COLOR:
+                textArea.setBackground(event.getColor());
+                break;
+
+            case Theme.EDITOR_SELECTED_FOREGROUND_COLOR:
+                textArea.setSelectedTextColor(event.getColor());
+                break;
+
+            case Theme.EDITOR_SELECTED_BACKGROUND_COLOR:
+                textArea.setSelectionColor(event.getColor());
+                break;
+        }
+    }
+
+    /**
+     * Receives theme font changes notifications.
+     */
+    public void fontChanged(FontChangedEvent event) {
+        if (event.getFontId() == Theme.EDITOR_FONT) {
+            textArea.setFont(event.getFont());
+        }
+    }
```

Comment:
did you format the file? it's difficult to see the changes here - could you please separate out the unrelated changes?

Reply:
yup, sometimes I cannot resist :)

Reply:
I've forced pushed two commits - 1st with reformat and 2nd with actual change